### PR TITLE
Charlesmchen/verification state change messages

### PR DESCRIPTION
**Pull request by charlesmchen**
### This PR is now for review.

PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
This PR is now for review.

PTAL @michaelkirk

**Comment by charlesmchen** [comment]
screen shot 2017-06-07 at 4 58 35 pm
screen shot 2017-06-07 at 4 46 53 pm

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 1]
```diff
@@ -1176,19 +1175,12 @@ - (BOOL)showSafetyNumberConfirmationIfNecessaryWithConfirmationText:(NSString *)
...
     // Ensure keyboard isn't hiding the "safety numbers changed" interaction when we
     // return from FingerprintViewController.
     [self dismissKeyBoard];
 
-    OWSFingerprintBuilder *builder =
-        [[OWSFingerprintBuilder alloc] initWithStorageManager:self.storageManager contactsManager:self.contactsManager];
-    OWSFingerprint *fingerprint =
-        [builder fingerprintWithTheirSignalId:theirSignalId theirIdentityKey:theirIdentityKey];
-
-    NSString *contactName = [self.contactsManager displayNameForPhoneIdentifier:theirSignalId];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: I've DRY'd up this logic by pulling it into `FingerprintViewController`.  It also has the effect of ensuring that we're always showing the latest identity in that view.

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 2]
```diff
@@ -2290,6 +2270,9 @@ - (void)handleInfoMessageTap:(TSInfoMessage *)message
         case TSInfoMessageTypeDisappearingMessagesUpdate:
             [self showConversationSettings];
             return;
+        case TSInfoMessageVerificationStateChange:
+            [self showFingerprintWithRecipientId:((OWSVerificationStateChangeMessage *)message).recipientId];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: Tapping on a verification state change message shows fingerprint view.

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: 👍

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m (hunk 2 above)
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: 👌

File: Signal/src/ViewControllers/FingerprintViewController.m [hunk 3]
```diff
@@ -20,34 +21,46 @@
...
-@property (strong, nonatomic) IBOutlet UIImageView *privacyVerificationQRCode;
-@property (strong, nonatomic) IBOutlet OWSHighlightableLabel *privacyVerificationFingerprint;
-@property (strong, nonatomic) IBOutlet UILabel *instructionsLabel;
-@property (strong, nonatomic) IBOutlet UIButton *scanButton;
+@property (nonatomic) TSStorageManager *storageManager;
+@property (nonatomic) OWSFingerprint *fingerprint;
+@property (nonatomic) NSString *contactName;
+@property (nonatomic) OWSQRCodeScanningViewController *qrScanningController;
+
+@property (nonatomic) IBOutlet UINavigationBar *modalNavigationBar;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/FingerprintViewController.m]: shucks, I was hoping if I dragged my feet you'd turn this into a programmatic VC in no time.

**Pull request merged**